components/script/dom/htmlmediaelement.rs

```diff
@@ -1280,7 +1406,43 @@ impl HTMLMediaElement {
                 //            https://www.w3.org/TR/media-frags/#media-fragment-syntax
                 //            https://github.com/servo/media/issues/156
 
-                // XXX Steps 12 and 13 require audio and video tracks support.
+                // Step 12
+                if !self.AudioTracks().is_empty() && self.AudioTracks().enabled_index().is_none() {
+                    let window = window_from_node(self);
+                    let this = Trusted::new(self);
+
+                    let _ = window
+                        .task_manager()
+                        .media_element_task_source()
+                        .queue(
+                            task!(media_track_change: move || {
+                                let this = this.root();
+                                this.AudioTracks().set_selected(0, true);
+                                this.upcast::<EventTarget>().fire_event(atom!("change"));
+                            }),
+                            window.upcast(),
+                        )
+                        .unwrap();
+                }
+
+                // Step 13
+                if !self.VideoTracks().is_empty() && self.VideoTracks().selected_index().is_none() {
```

Comment:
Steps 12 and 13 should be calling set_selected/set_enabled only; the DOM event code should be part of that implementation per previous comments.